apps/sms/views/shared/js/system_message_handler.js

```diff
@@ -329,6 +332,12 @@
       // Conversation view to the user, we remove that notification from the
       // tray that causes the second system message with "clicked" set to false.
       if (!notification.clicked || !notification.data) {
+        // Force closing the window for notification removal case
+        // because navigation didn't init correctly if pending notification
+        // message existed.
+        if (Navigation.hasPendingInit()) {
+          return Promise.reject(window.close());
```

Comment:
like before, I don't get whi you reject with the result of window.close ?

why not doing this:
```javascript
window.close();
return Promise.reject(new Error('The notification has been dismissed.'));
```